src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs

```diff
@@ -81,22 +89,59 @@ public static void X509Cert2Test()
 
 #if netstandard17
         [Fact]
-        public static void TestVerify()
+        [OuterLoop("May require using the network, to download CRLs and intermediates")]
+        public void TestVerify()
         {
+            bool success;
+
             using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
             {
                 // Fails because expired (NotAfter = 10/16/2016)
-                Assert.False(microsoftDotCom.Verify());
+                Assert.False(microsoftDotCom.Verify(), "MicrosoftDotComSslCertBytes");
             }
 
             using (var microsoftDotComIssuer = new X509Certificate2(TestData.MicrosoftDotComIssuerBytes))
             {
-                Assert.True(microsoftDotComIssuer.Verify()); // NotAfter=10/31/2023
+                // NotAfter=10/31/2023
+                success = microsoftDotComIssuer.Verify();
+                if (!success)
+                {
+                    LogVerifyErrors(microsoftDotComIssuer, "MicrosoftDotComIssuerBytes");
+                }
+                Assert.True(success, "MicrosoftDotComIssuerBytes");
             }
 
             using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
             {
-                Assert.True(microsoftDotComRoot.Verify()); // NotAfter=7/17/2036
+                // NotAfter=7/17/2036
+                success = microsoftDotComRoot.Verify();
+                if (!success)
+                {
+                    LogVerifyErrors(microsoftDotComRoot, "MicrosoftDotComRootBytes");
+                }
+                Assert.True(success, "MicrosoftDotComRootBytes");
+            }
+        }
+
+        private void LogVerifyErrors(X509Certificate2 cert, string testName)
+        {
+            // Emulate cert.Verify() implementation in order to capture and log errors.
+            try
+            {
+                using (var chain = new X509Chain())
+                {
+                    if (!chain.Build(cert))
+                    {
+                        foreach (X509ChainStatus chainStatus in chain.ChainStatus)
+                        {
+                            _log.WriteLine(string.Format($"X509Certificate2.Verify error: {testName}, {chainStatus.Status}, {chainStatus.StatusInformation}"));
+                        }
+                    }
+                }
```

Comment:
You may want to log if Build returns true (which could happen if a networking error didn't happen this time)